master/internal/configpolicy/task_config_policy.go

```diff
@@ -244,50 +249,51 @@ func findAllowedPriority(scope *int, workloadType string) (limit int, exists boo
 			var configs model.CommandConfig
 			err = json.Unmarshal([]byte(*configPolicies.InvariantConfig), &configs)
 			if err != nil {
-				return 0, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
+				return nil, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
 			}
 			if configs.Resources.Priority != nil {
 				adminPriority := *configs.Resources.Priority
-				return adminPriority, false,
+				return ptrs.Ptr(adminPriority), false,
 					fmt.Errorf("priority set by invariant config: %w", errPriorityImmutable)
 			}
 		case model.ExperimentType:
 			var configs expconf.ExperimentConfigV0
 			err = json.Unmarshal([]byte(*configPolicies.InvariantConfig), &configs)
 			if err != nil {
-				return 0, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
+				return nil, false, fmt.Errorf("unable to unmarshal task config policies: %w", err)
 			}
 			if configs.RawResources != nil && configs.RawResources.RawPriority != nil {
 				adminPriority := *configs.RawResources.RawPriority
-				return adminPriority, false,
+				return ptrs.Ptr(adminPriority), false,
```

Comment:
```suggestion
				return &adminPriority, false,
```
just take an address